File: week-02/day-01/draw_diagonal.py
# Write a program that reads a number from the standard input, then draws a
# square like this:
#
#
# %%%%%
# %%  %
# % % %
# %  %%
# %   %
# %%%%%
#
# The square should have as many lines as the number was
a = int(input("Please enter a number: "))
for i in range(1,a+1):
    if i == 1:
        print("%"*(a-1))
    elif i < a-1:
        print("%"+" "*(i-2)+"%"+" "*(a-2-i)+"%")
    elif i < a:
        print("%"+" "*(a-3)+"%")
    else:
        print("%"*(a-1))

File: week-07/day-02/draw_diagonal.js
'use strict';

let lineCount = 11;


// Write a program that draws a
// square like this:
//
//
// %%%%%
// %%  %
// % % %
// %  %%
// %   %
// %%%%%
//
// The square should have as many lines as lineCount is

for (let i = 1; i < lineCount+1; ++i) {
    if (i === 1 || i === lineCount) {
        console.log("%".repeat(lineCount-1));
    } else if (i < lineCount-1) {
        console.log("%" + " ".repeat(i-2) + "%" + " ".repeat(lineCount-2-i) + "%");
    } else {
        console.log("%" + " ".repeat(lineCount-3) + "%");
    }
}
